pkg/apis/pipeline/v1alpha1/git_resource.go

```diff
@@ -30,6 +34,25 @@ type GitResource struct {
 	ServiceAccount string `json:"serviceAccount,omitempty"`
 }
 
+// NewGitResource create a new git resource to pass to Knativve Build
+func NewGitResource(r *PipelineResource) *GitResource {
+	gitResource := GitResource{
+		Name: r.Name,
+		Type: r.Spec.Type,
```

Comment:
Can we check if the type is `PipelineResourceTypeGit` and if not throw an error

Reply:
I see you do check below in AddInputResource before creating it, however, i still think having it here is useful.
We don't want users accidentally calling NewGitResource on non git type resources

Reply:
ok, will add it